src/Resources/Resources.Test/ScenarioTests/RoleAssignmentTests.cs

```diff
@@ -174,7 +174,7 @@ public void RaGetByUPNWithExpandPrincipalGroups()
             TestRunner.RunTestScript("Test-RaGetByUPNWithExpandPrincipalGroups");
         }
 
-        [Fact]
+        [Fact(Skip = "Breaking PR in 2 parts needs this test to be disabled temporarily")]
```

Comment:
The change I'm making changes the codeflow intended for when creating role assignments through an SP (e.g. Azure app)
The flow is not broken but it does need to be updated. That will be done on a separate PR as it might be an extensive change by itself.

Reply:
Sounds OK, but what about the above cases with `Skip = "Test is not deterministic, only works on record mode"`?

Reply:
the tests themselves pass, in record mode they use my credentials and my tenant and everything goes well ✅
the problem is that the tests generate guids on the fly which break the tests on playback mode. When trying to read the guid from the session records, since it always change it's impossible to request the same guid twice.